vault/hcp_link/capabilities/link_control/link_control.go

```diff
@@ -110,7 +111,14 @@ func (h *hcpLinkControlHandler) Stop() error {
 	return nil
 }
 
-func (h *hcpLinkControlHandler) PurgePolicy(ctx context.Context, req *link_control.PurgePolicyRequest) (*link_control.PurgePolicyResponse, error) {
+func (h *hcpLinkControlHandler) PurgePolicy(ctx context.Context, req *link_control.PurgePolicyRequest) (retResp *link_control.PurgePolicyResponse, retErr error) {
+	defer func() {
+		if r := recover(); r != nil {
+			h.logger.Error("panic serving purge policy request", "error", r, "stacktrace", string(debug.Stack()))
+			retErr = fmt.Errorf("internal server error")
```

Comment:
Should `ErrInternalError` be used instead?

Reply:
That's a great idea. Will do, thanks!